crates/core/src/val/value/convert/cast.rs

```diff
@@ -906,6 +906,10 @@ impl Value {
 
 	/// Try to convert this value to a `Geometry` of a certain type
 	fn cast_to_geometry(self, val: &[GeometryKind]) -> Result<Geometry, CastError> {
+		// An array of two numbers can be cast into a point
+		if let Some(p) = Geometry::array_to_point(&self) {
+			return Ok(Geometry::Point(p));
+		}
```

Comment:
This will (if i read it correctly) make a cast to any geometry kind return a point if it can be cast to a point. So `<polygon> [1.0, 2.0]` for example, will return a point.